.github/workflows/update-docs-on-docs-commit.yml

```
@@ -0,0 +1,42 @@
+# This workflow is centrally managed in https://github.com/asyncapi/.github/
+# Don't make changes to this file in this repo as they will be overwritten with changes made to the same file in above mentioned repo
+
+# The given workflow is responsible for generating docs and creating PR with them when there is a commit with docs: prefix
+
+# This workflow will be updated in all repos with the topic autogenerate-docs
+
+name: 'Update generated parts of documentation on docs: commits'
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  docs-gen:
+    name: 'Generate docs and create PR'
+    runs-on: ubuntu-latest
+    # PR should be created within this GH action only if it is a docs: commit
+    # Otherwise it will conflict with release workflow
+    if: startsWith(github.event.commits[0].message, 'docs:')
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: Install dependencies
+        run: npm ci
+      - name: Regenerate docs
+        run: npm run generate:assets
+      - name: Create Pull Request with updated docs
+        uses: peter-evans/create-pull-request@v5
```

```suggestion
        uses: peter-evans/create-pull-request@153407881ec5c347639a548ade7d8ad1d6740e38 # use 4.2.4 https://github.com/peter-evans/create-pull-request/releases/tag/v5.0.2
```

it is a good practice that in case of not official GitHub Actions, to use commits instead of tags as they are not vulnerable to any bad actors activity (removal of tag, or recreating tag that points to code that sneaks out the api keys)

[reference](https://github.com/asyncapi/community/pull/805#discussion_r1272500392)

---

Thanks for the heads up. I will do so.

---

You're welcome! :grin: